Automattic/wp-calypso
# wp-admin Site Default: refresh admin sidebar menu after saving the interface option

- Closes https://github.com/Automattic/dotcom-forge/issues/4157

## Proposed Changes

* Refreshes the admin sidebar menu when the site admin interface option changes
* It adds a new spinner inside the menu items

Another solution, more straightforward would be changing the menu more silently.

## Testing Instructions



* Use this branch or calypso live
* On an atomic site, navigate to `Settings` > `Hosting configuration` : `/hosting-config/$siteSlug`
* Scroll down until you find: "Admin infterface style"
* Change the option inside the card
* Observe the menu on the left displays a spinner and after few seconds it displays the new menu pointing to the right place. (`wp-admin` vs `calypso`)

https://github.com/Automattic/wp-calypso/assets/779993/34d4e474-500c-4a5f-848e-6d646e2546a4

## Pre-merge Checklist
